### PR TITLE
Avoid version conflict and lint issues in host application

### PR DESCRIPTION
**Pull request by guperrot**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

* Use host app versions for various android settings when possible to avoid lint errors.
* Also use latest version of Firebase and let react native pull support lib to avoid conflicts.
* Verified lint and runtime of TestApp.
* Verified lint and runtime of new blank app using RN 0.57.4
* Verified publishing 1.9.0-0 temp version to bintray snapshot with the change for shared android module and consumed that version from TestApp.

## Related PRs or issues

#451 and #365

## Misc

Not sure about the impact of older versions of RN yet.

### CODE REVIEW

**Comment by guperrot** [comment]
The push part of the fix works only for recent react native versions. Need to find another solution for push gradle.

**Comment by guperrot** [comment]
We could document to add recent support libraries explicitly for old react native versions.

**Comment by guperrot** [comment]
Could not fully automate all lint fixes but now push works without missing dependencies and without lint errors.

To fix lint errors at `:app:lint` level caused by Firebase VS React Native, the application needs to manually add the dependencies of support libraries that mismatch versions. Only customers that want to resolve lint issue will need extra steps in `app/build.gradle`.

Still need testing on old RN versions.

File: AppCenterReactNativeShared/android/build.gradle [hunk 1]
```diff
@@ -109,7 +111,6 @@ allprojects {
 }
 
 dependencies {
-    compile 'com.android.support:appcompat-v7:23.0.1'
```
Comment (dhei) [inline AppCenterReactNativeShared/android/build.gradle]: 👍 I like removing this -- it shouldn't be there in the first place.

File: AppCenterReactNativeShared/android/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,5 @@
-#Tue Jul 17 18:39:49 PDT 2018
```
Comment (dhei) [inline AppCenterReactNativeShared/android/gradle/wrapper/gradle-wrapper.properties]: nit-picking, use Android Studio to auto upgrade this.

Comment (dhei) [reply AppCenterReactNativeShared/android/gradle/wrapper/gradle-wrapper.properties]: ok, never mind

**Comment by guperrot** [comment]
It still works at runtime when using this patch on RN 0.34 / 0.50. Lint will work only on latest RN version though (0.57.4 as of now).

**Pull request merged**